languages/java/oso/src/main/java/com/osohq/oso/Query.java

```diff
@@ -76,7 +109,27 @@ private void handleCall(
       ffiQuery.callResult(callId, null);
       return;
     }
+  }
+
+  /** Helper for `NextExternal` query events */
+  private void handleNextExternal(long callId, JSONObject iterable) throws Exceptions.OsoException {
+    if (!calls.containsKey(callId)) {
+      Object result = host.toJava(iterable);
+      Enumeration<Object> enumResult;
```

Comment:
@samscott89 I see there was already some usage of `Enumeration<Object>` in the previous code, but did you think about using Iterator instead?

https://docs.oracle.com/javase/10/docs/api/java/util/Iterator.html

> An iterator over a collection. **Iterator takes the place of Enumeration in the Java Collections Framework.** Iterators differ from enumerations in two ways:
>    - Iterators allow the caller to remove elements from the underlying collection during the iteration with well-defined semantics.
>    - Method names have been improved.